lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -1759,16 +1760,25 @@ bool IRGenDebugInfoImpl::lineEntryIsSane(SILLocation::DebugLoc DL,
   // All bets are off for optimized code.
   if (!VerifyLineTable || Opts.shouldOptimize())
     return true;
-  // We entered a new lexical block.
+
+  // We entered a new lexical block, reset the last
+  // location we keep track of.
   if (DS != LastScope)
-    PreviousLineEntries.clear();
-  if (DL.Line == 0 || DL == PreviousDebugLoc)
+    PreviousDebugLoc = DebugLocKey();
+
+  // Skip line entry zero.
+  if (DL.Line == 0)
+    return true;
+
+  auto DLK = DebugLocKey(DL);
+
+  if (DLK >= PreviousDebugLoc || !DLK.sameFile(PreviousDebugLoc.second)) {
```

Comment:
Can you add a comment that explains that this verifies that line numbers are monotonically increasing within basic blocks?